xbmc/platform/android/CPUInfoAndroid.cpp

```diff
@@ -31,3 +57,23 @@ CCPUInfoAndroid::CCPUInfoAndroid()
   if (CAndroidFeatures::HasNeon())
     m_cpuFeatures |= CPU_FEATURE_NEON;
 }
+
+float CCPUInfoAndroid::GetCPUFrequency()
+{
+  float freq = 0.f;
+
+  if (!m_pPosixFile)
+    return freq;
+
+  if (m_pPosixFile->Open(CURL("/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq")))
```

Comment:
Same here. Open with another URL will basically call reopen. It'd be a bit more clean if file was a local variable

Reply:
This is the method that is called at periodic intervals and is used previously obtained `CPosixFile` instance.